ark/phenotyping/som_utils.py

```diff
@@ -246,32 +381,40 @@ def compute_cell_cluster_channel_avg(pixel_channel_avg, cell_counts,
 
     # add columns back
     meta_cols = ['cell_size', 'fov', 'segmentation_label']
-    weighted_cell_channel_avg[meta_cols] = cell_counts[meta_cols]
+    weighted_cell_channel_avg[meta_cols] = cell_counts_sub[meta_cols]
```

Comment:
One more thing. When I was testing by subsetting on FOVs, this line caused an error. If cell_counts_sub is subsetted for FOVs in line 338 above, the indices of this table changes. The indices of the two tables no longer match. So this line returns NaN for all the meta columns. Need to reset the indices before doing this line.

Reply:
Yeah, the multitude of new indices that get created can be a real pain to keep track of. I think just to ensure everything is standardized I'll reset the indices anytime subsetting happens.

Reply:
So actually on an unrelated note, when computing the weighted channel average per cell we're using the _z-scored_ average channel values per desired pixel cluster. Is this what we were going for or did we want the un-z-scored average channel values? If the latter, I'll have to add code to save that in `example_pixel_clustering.ipynb`.

Reply:
Oh this is a good point. I think we should use the un-zscored average channel values.

Reply:
Remember before when you had the z-scoring in both the python and R code? Perhaps we can just leave it in the R code and save the un-zscored values in python, instead of saving both.

Reply:
I think before I move forward, we should first re-clarify which parts of the pipeline need to be z-scored and which should not be z-scored. So far, this is my understanding:

1. Visualization for average channel expression per pixel SOM and pixel meta cluster: I thought this was supposed to be z-scored, but does Zak's tool need the un-z-scored version of this?
2. Inputs for pixel consensus clustering: z-scored average channel expression per pixel SOM cluster
3. Inputs for weighted channel expression per cell computation: un-z-scored average channel expression per pixel SOM or pixel meta cluster (plus the **_un-normalized_** number of desired pixel cluster type per cell)
4. Visualization for average number of desired pixel cluster type per cell SOM and cell meta cluster: like 1, I thought this was supposed to be z-scored, but does Zak's tool need the un-z-scored version of this?
5. Inputs for cell consensus clustering: z-scored average number of desired pixel cluster type per cell SOM cluster
6. Visualization for average weighted channel expression per cell (this one's the one we really need to get right): my previous understanding was that it should be z-scored averages per cell SOM or cell meta cluster on the data from part 3 (which was computed using **_un-z-scored_** data). However, based on your comments below, it seems that we should actually not add z-scoring to the average weighted channel expression per cell SOM or meta cluster?

Making these changes shouldn't be hard, about a day's work at most (might need to spread it out over 2 depending on how other projects progress), just want to make absolute certain nothing else fell through the cracks beforehand.

Reply:
1. **Visualization for average channel expression per pixel SOM and pixel meta cluster**: Yes, Zak's tool needs un-zscored data because when you re-assign clusters to metaclusters, the z-scores will change. So Zak's tool does the z-scoring.
2. **Inputs for pixel consensus clustering**: Z-scored (so I think you can do this in R right before the clustering).
3. **Inputs for weighted channel expression per cell computation**: Yes, un-zscored average channel expression per pixel cluster and un-normalized number of pixels per cell (multiply these two). But after the matrix multiplication, these numbers should be normalized by cell size before making the heatmaps for average weighted channel expression per cell (#6) - I'm not sure if you're doing this right now.
4. **Visualization for average number of desired pixel cluster type per cell SOM and cell meta cluster**: We probably won't have Zak's tool run on this data, as the default. What's important is the cell cluster x marker expression table when doing the manual reclustering. The cell cluster x pixel cluster will probably just be an informative visualization for the user. Probably better to save the un-zscored version of this data (just in case the user wants to run Zak's tool on it, then they can load this table and run it themselves).
5. **Inputs for cell consensus clustering**: Z-scored (so again, I think you can do this in R right before clustering).
6. **Visualization for average weighted channel expression per cell**: Un-zscored.

Reply:
Ok, it should be good to go now!